Requests timeout and retrying. Auth service

YFatMR opened the pull request:
Added timeout for http servers and
database operations
Implemented auth service
Implemented auth unary interceptor
Added common logick of microservices to core lib
Added mongo init rerty policy
Added grpcc lints to core lib (with keepalive connection)

YFatMR committed:
feat: Added timeout for http servers and
database operations

YFatMR committed:
feat: Added retry policy for grpc dial

YFatMR committed:
fix: Added protoc installation for CI

YFatMR committed:
fix: Delete local GOPATH from script

YFatMR committed:
fix: Added raw build for CI

VladimirBalun commented on .env:
```diff
@@ -9,7 +9,7 @@ USER_SERVICE_MONGODB_USERNAME=admin
 USER_SERVICE_MONGODB_DATABASE_NAME=users
 USER_SERVICE_MONGODB_DATABASE_COLLECTION_NAME=users
-USER_SERVICE_MONGODB_CONNECTION_TIMEOUT_SECONDS=10
+USER_SERVICE_MONGODB_OPERATION_TIMEOUT_SECONDS=1
```
Если в какой-то момент придется mongo заменить - придется менять имена переменных и внутри кода тоже нейминги менять придется, может просто писать USER_SERVICE_DATABASE_COLLECTION_NAME и типо такого?

YFatMR replied:
согласен, поправил

YFatMR replied:
согласен, поправил

VladimirBalun commented on front_service/cmd/main.go:
```diff
@@ -2,20 +2,27 @@ package main
+	"google.golang.org/grpc/credentials/insecure"
 )
 
 func main() {
```
main уже начинает получаться большим я тяжелым, может на init ф-ии разбить и вынести в отдельный файл init.go?

YFatMR replied:
Чуть позже разобью, пока только вынес инициализацию монги отдельно и уменьшил строк 10, постепенно сделаю все красиво

VladimirBalun commented on front_service/internal/user_server/server.go:
```diff
@@ -8,29 +8,32 @@ import (
-	return &frontUserServer{
+func NewFrontUserServer(userServerAddress string, logger *loggers.OtelZapLoggerWithTraceID,
+	tracer trace.Tracer, backoffConfig backoff.Config,
+) *FrontUserServer {
```
Можно по значению возвращать, чтобы не через кучу шло

VladimirBalun commented on front_service/internal/user_server/server.go:
```diff
@@ -58,6 +66,11 @@ func (s *frontUserServer) GetUserByID(ctx context.Context, request *proto.UserID
 		s.userServerAddress,
```
На каждом запросе будет создавать конекшен, а можно создать аля клиента, у которого будет постоянное соединение и к нему могут приходить различные запросы из очереди на выполнение, это как вариант

YFatMR replied:
Так и сделал, вынес в core/pkg/grpc_clients

VladimirBalun commented on user_service/cmd/main.go:
```diff
@@ -28,7 +30,7 @@ import (
 )
 
 func main() {
```
Тоже самое с main-ом

YFatMR replied:
пофикшу позже

VladimirBalun replied:
Видимо забыл это сделать)

VladimirBalun commented on user_service/cmd/main.go:
```diff
@@ -28,7 +30,7 @@ import (
 
 func main() {
-	time.Sleep(5 * time.Second)
+	time.Sleep(3 * time.Second)
```
А зачем мы спим в начале?

YFatMR replied:
Сделал ретрай полиси к базе данных, теперь спать не нужно (раньше ждали базку)

YFatMR committed:
feat:
Implemented auth service
Implemented auth unary interceptor
Added common logick of microservices to core lib
Added mongo init rerty policy
Added grpcc lints to core lib (with keepalive connection)

YFatMR committed:
fix: Fixed linter

VladimirBalun commented on user_service/internal/entities/user_id.go:
```diff
@@ -0,0 +1,28 @@
+	}
+}
+
+func NewUserIDFromProtobuf(userID *proto.UserID) (*UserID, error) {
```
Все-таки реши в entities оставить?)

VladimirBalun commented on user_service/internal/entities/user.go:
```diff
@@ -0,0 +1,38 @@
+	return newUser(name, surname)
+}
+
+func NewMockUser(name string, surname string) *User {
```
Моки, кажется, тоже не стоит сюда добавлять

VladimirBalun commented on auth_service/cmd/main.go:
```diff
@@ -0,0 +1,161 @@
+)
+
+func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
```
Тоже самое с gracefull shutdown
